the-collab-lab/tcl-36-smart-shopping-list
# Refactored listView layout, logic, and styling

## Description

Made header and footer position 'sticky' to allow flex to work correctly ('fixed' was fixing to the viewport and negating flex parent relationships). I think I caught all the remaining vanilla CSS and replaced with tailwind. 

Refactored item display from list items in grid, to rows in a table. Now items are more visible and more attractively aligned. Added Lindsey's toggle functionality with details (__how had I never seen this before?!__) to each table so that user can collapse each section's list independently. Updated logic for empty groups so that no table or toggle is displayed, just the line "No items needed in this time frame".

Set save and search features in their own container and made it sticky just below the header so that user has access to those features regardless of how long their lists become.

TODO: We weren't thrilled with the scrolling effect of this container and discussed placing this in a sidebar or adding to the header when in listView. This task will be added as its own issue.

## Related Issue
Closes #42 
Closes #46
Closes #47

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|    | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

![image](https://user-images.githubusercontent.com/81432121/156252396-4573edd9-187c-4053-95a6-ab954c4e255d.png)

### After

![image](https://user-images.githubusercontent.com/81432121/156253466-5ec992a5-3577-4bd9-ac44-3e7b45c9964b.png)

## Testing Steps / QA Criteria

Pull down this branch to your local environment. Clear any existing token from your local storage with ```localStorage.clear()``` to see features beginning at the homescreen and welcome page. If you would like to see the exact sorting from the screenshots, enter the token ```gnat paulo amigo```
